video_stream_with_annotations.py

```diff
@@ -151,7 +151,7 @@ def calculate_font_size(self, dialogue_box_width, dialogue_box_height, text, ini
 
         def fits(font_size):
             try:
-                font = ImageFont.truetype(font_path, font_size)
+                font = ImageFont.truetype(self.font_path, font_size)
```

Comment:
you shouldn't be loading a font constantly. the old code  loaded the font once. its very expensive and usually you run out of file handles